addons/project/populate/project.py

```diff
@@ -57,15 +57,43 @@ def get_stage_ids(random, **kwargs):
 class ProjectTask(models.Model):
     _inherit = "project.task"
     _populate_sizes = {"small": 500, "medium": 5000, "large": 50000}
-    _populate_dependencies = ["project.project"]
+    _populate_dependencies = ["project.project", "res.partner", "res.users"]
 
     def _populate_factories(self):
         project_ids = self.env.registry.populated_models["project.project"]
         stage_ids = self.env.registry.populated_models["project.task.type"]
+        user_ids = self.env.registry.populated_models['res.users']
+        partner_ids_per_company_id = {
+            group['company_id'][0]: group['ids']
+            for group in self.env['res.partner']._read_group(
+                domain=[('company_id', '!=', False), ('id', 'in', self.env.registry.populated_models["res.partner"])],
+                fields=['company_id', 'ids:array_agg(id)'],
+                groupby=['company_id'],
```

Comment:
No need to name the parameters if the parameters are given in the right orders. 🙂 
```suggestion
                [('company_id', '!=', False), ('id', 'in', self.env.registry.populated_models["res.partner"])],
                ['company_id', 'ids:array_agg(id)'],
                ['company_id'],
```